### PR TITLE
Bala/Flutter charts integration setup

### PR DESCRIPTION
**Pull request by balakrishna-deriv**
Features added:

- Chart type - Hollow, OHLC
- Crosshair for web
- Tooltip for web
- Light theme
- Indicators bug fixes
- Indicators - Edit, Delete, Move up & down, Expand & Collapse.
- Barriers & Markers - Option and Accumulator contract types
- Download the chart as a png or csv
- Save template
- Movement of drawing tools
- Draggable barrier
- Contract replay - Barriers and Markers for all contract types

### CODE REVIEW

File: src/Constant.tsx [hunk 1]
```diff
@@ -107,80 +68,80 @@ export const drawTools: TDrawTools = {
...
     {
-        id: 'momentum',
+        category: 'momentum',
         name: t.translate('Momentum'),
         icon: IndicatorCatMomentumIcon,
         items: [
             {
-                id: 'Awesome',
-                name: t.translate('Awesome Oscillator'),
+                name: 'AwesomeOscillator',
```
Comment (maryia-deriv) [inline src/Constant.tsx:76]: @balakrishna-deriv I noticed that indicators names here vary greatly: some are written in camelCase, some in snake_case, some in lower case, some in upper case. Is there any naming convention for these?

File: src/Constant.tsx [hunk 2]
```diff
@@ -333,113 +294,971 @@ export const getIndicatorsTree = (): TIndicatorsTree[] => [
...
+    fcb: FractalChaosBandIndicatorConfig,
 };
 
 export const ChartTypes = [
-    { id: 'mountain', text: t.translate('Area'), candleOnly: false, icon: TypeAreaIcon },
-    { id: 'candle', text: t.translate('Candle'), candleOnly: true, icon: TypeCandleIcon },
-    { id: 'hollow_candle', text: t.translate('Hollow'), candleOnly: true, icon: TypeHollowIcon },
-    { id: 'colored_bar', text: t.translate('OHLC'), candleOnly: true, icon: TypeOhlcIcon },
+    { id: 'line', text: t.translate('Area'), candleOnly: false, icon: TypeAreaIcon },
+    { id: 'candles', text: t.translate('Candle'), candleOnly: true, icon: TypeCandleIcon },
```
Comment (maryia-deriv) [inline src/Constant.tsx:1246]: @balakrishna-deriv what's the main reason to rename the ids? is it only for consistency between smartcharts and flutter-charts?

File: src/store/BottomWidgetsContainerStore.ts [hunk 3]
```diff
@@ -36,46 +30,8 @@ export default class BottomWidgetsContainerStore {
...
-                this.stx.chart.yAxis.initialMarginBottom = marginBottom;
-                this.stx.calculateYAxisMargins(this.stx.chart.panel.yAxis);
-                this.stx.draw();
-            }
-            if (!this.mainStore.state.shouldMinimiseLastDigits) {
-                this.mainStore.state.setShouldMinimiseLastDigit(this.stx.chart.panel.height < 460);
-            }
-        }
-    };
+    updateChartMargin = (): void => {};
```
Comment (maryia-deriv) [inline src/store/BottomWidgetsContainerStore.ts:36]: @balakrishna-deriv just curious, why is it empty now? 🙂

File: src/store/ChartState.ts [hunk 4]
```diff
@@ -453,251 +374,62 @@ class ChartState {
             (this.chartType && this.mainStore.chartType.isTypeCandle(this.chartType));
 
         if (this.context && isTimeUnitSecond && isChartTypeCandle) {
-            this.setChartType('mountain'); // if granularity is zero, set the chartType to 'mountain'
+            this.setChartType('line'); // if granularity is zero, set the chartType to 'mountain'
```
Comment (maryia-deriv) [inline src/store/ChartState.ts:377]:
```suggestion
            this.setChartType('line'); // if granularity is zero, set the chartType to 'line'
```
@balakrishna-deriv nit-pick🙂

File: src/store/ChartStore.ts [hunk 5]
```diff
@@ -54,7 +48,7 @@ type TSpan = {
     maintainPeriodicity?: boolean;
     padding?: number;
     forceLoad?: boolean;
-    chart?: typeof CIQ.ChartEngine.Chart;
+    chart?: any;
```
Comment (maryia-deriv) [inline src/store/ChartStore.ts:51]: @balakrishna-deriv Are the `chart` variable and `TRange` and `TSpan` types still used anywhere? it seems like they are no longer used in this file.

**Review by akmal-deriv — DISMISSED** [review]

File: src/store/ChartStore.ts (hunk 5 above)
Comment (balakrishna-deriv) [reply src/store/ChartStore.ts]: @maryia-deriv I've removed those. Thanks for checking 🙏

File: src/store/BottomWidgetsContainerStore.ts (hunk 3 above)
Comment (balakrishna-deriv) [reply src/store/BottomWidgetsContainerStore.ts]: Removed it

File: src/Constant.tsx (hunk 2 above)
Comment (balakrishna-deriv) [reply src/Constant.tsx:1269]: Yes, to make it consistent between smartcharts and flutter-charts

File: src/Constant.tsx (hunk 1 above)
Comment (balakrishna-deriv) [reply src/Constant.tsx]: These are the ids in flutter-charts. Yes, they are inconsistent but I don't want to change them in the flutter-chart.
I've renamed the `name` prop to `flutter-chart-id`.

**Review by maryia-deriv — DISMISSED** [review]

**Review by yashim-deriv — APPROVED** [review]

**Pull request merged**